front/src/modules/timesStops/utils.ts

```diff
@@ -25,11 +25,28 @@ export const formatSuggestedViasToRowVias = (
       formattedOps.unshift(operationalPoints[originIndexInOps]);
     }
   }
-  return formattedOps.map((op, i) => ({
-    ...op,
-    name: op.name || t('waypoint', { id: op.opId }),
-    isMarginValid: op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true,
-    onStopSignal: op.onStopSignal || false,
-    arrival: i === 0 ? startTime?.substring(11, 19) : op.arrival,
-  }));
+  return formattedOps.map((op, i) => {
+    if (!('uic' in op)) {
+      const correspondingOp = pathSteps.find((step) => step.id === op.opId);
+      if (correspondingOp) {
+        return {
+          ...op,
+          name: op.name || t('waypoint', { id: op.opId }),
+          isMarginValid: op.theoreticalMargin
+            ? marginRegExValidation.test(op.theoreticalMargin)
+            : true,
+          onStopSignal: correspondingOp.onStopSignal || false,
+          arrival: i === 0 ? startTime?.substring(11, 19) : correspondingOp.arrival,
+          stopFor: correspondingOp.stopFor,
+          theoreticalMargin: correspondingOp.theoreticalMargin,
+        };
+      }
+    }
+    return {
+      ...op,
+      isMarginValid: op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true,
+      onStopSignal: op.onStopSignal || false,
+      arrival: i === 0 ? startTime?.substring(11, 19) : op.arrival,
+    };
```

Comment:
instead of nesting condition and give 2 returns, can't you manage `isMarginValid` , `onStopSignal` and `arrival` then return the result?

Reply:
I could propose you that but I think I prefer the first syntax. Do you have another idea ? 

```
  return formattedOps.map((op, i) => {
    let opToFormat = {
      ...op,
      isMarginValid: op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true,
      onStopSignal: op.onStopSignal || false,
      arrival: i === 0 ? startTime?.substring(11, 19) : op.arrival,
    };
    if (!('uic' in op)) {
      const correspondingOp = pathSteps.find((step) => step.id === op.opId);
      if (correspondingOp) {
        opToFormat = {
          ...opToFormat,
          name: op.name || t('waypoint', { id: op.opId }),
          arrival: i === 0 ? startTime?.substring(11, 19) : correspondingOp.arrival,
          stopFor: correspondingOp.stopFor,
          theoreticalMargin: correspondingOp.theoreticalMargin,
        };
      }
    }

    return opToFormat;
  });
```

Reply:
Maybe something like that ?

```suggestion
      const correspondingOp = pathSteps.find((step) => step.id === op.opId);

    return {
      ...op,
      isMarginValid: op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true,
      onStopSignal: op.onStopSignal || false,
      arrival: i === 0 ? startTime?.substring(11, 19) : op.arrival,
      ...(correspondingOp && {
        name: op.name || t('waypoint', { id: op.opId }),
        stopFor: correspondingOp.stopFor,
        theoreticalMargin: correspondingOp.theoreticalMargin,
      }),
    };
```

Reply:
Alternatively, keep the current logic with 2 returns, and use a const for the `isMarginValid` field?

```js
const isMarginValid = op.theoreticalMargin ? marginRegExValidation.test(op.theoreticalMargin) : true;
if (correspondingOp) {
  return {
    ...op,
    name: op.name || t('waypoint', { id: op.opId }),
    isMarginValid,
    // etc
  };
}

return {
  ...op,
  isMarginValid,
  // etc
};
```

Reply:
that's the way I was thinking about

Reply:
I updated my suggestion ⏫

Reply:
I think for code readability/maintainability that it is still better with 2 returns. We have only one big condition between "has 'uic'" or not instead of a small condition for each properties which is harder to track which one comes from `op` or from `correspondingOp` . I can add @emersion suggestion to remove the repetition for isMarginValid.
Is that ok for you ?

Reply:
I disagree with both your readability/maintainability arguments, the solution proposed by @SarahBellaha seems to me cleaner and far more maintainable, anyway your solution and the one from @emersion are both functional, so feel free! it was just a suggestion